### PR TITLE
[controller] Update Go dependencies

### PR DESCRIPTION
**Pull request by AleksZimin**
## Description

This PR updates the Go dependencies used in the controller to the latest versions.

## Why do we need it, and what problem does it solve?

Updating Go dependencies ensures that the project benefits from the latest improvements, bug fixes, and security patches. It also helps maintain compatibility with other tools and libraries that may rely on updated versions.

## What is the expected result?

- Improved security and stability of the controller.
- Better compatibility with the latest tools and libraries.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -9,10 +9,15 @@ issues:
     - "exported: .*"
 
 linters-settings:
+  # gci write --skip-generated -s standard -s default -s 'prefix(sds-local-volume-controller,sds-local-volume-csi,sds-local-volume-scheduler-extender,webhooks)'  .
```
Comment (ViktorKram) [inline .golangci.yaml:12]: Do we really need this here?

File: images/sds-local-volume-scheduler-extender/src/pkg/controller/lvg_watcher_cache.go [hunk 2]
```diff
@@ -40,119 +40,103 @@ func RunLVGWatcherCacheController(
...
-			if !ok {
-				err = errors.New("unable to cast event object to a given type")
-				log.Error(err, "[RunLVGWatcherCacheController] an error occurred while handling create event")
-				return
-			}
-			cache.DeleteLVG(lvg.Name)
-			log.Debug(fmt.Sprintf("[RunLVGWatcherCacheController] LVMVolumeGroup %s was deleted from the cache", lvg.Name))
-		},
-	})
+	err = c.Watch(
```
Comment (ViktorKram) [inline images/sds-local-volume-scheduler-extender/src/pkg/controller/lvg_watcher_cache.go:43]: Lets write in one-line according to our other projects and code-style.

File: .golangci.yaml (hunk 1 above)
Comment (AleksZimin) [reply .golangci.yaml]: Fixed

File: images/sds-local-volume-scheduler-extender/src/pkg/controller/lvg_watcher_cache.go (hunk 2 above)
Comment (AleksZimin) [reply images/sds-local-volume-scheduler-extender/src/pkg/controller/lvg_watcher_cache.go]: Fixed

**Review by ViktorKram — APPROVED** [review]

**Pull request merged**